### PR TITLE
BC-2364 Display toast when trying to send mail to an already registered user

### PR DESCRIPTION
**Pull request by MateuszSapala**
# Short Description 
When an administrator/teacher tries to send an email to an already registered user, the corresponding toast is displayed

## Links to Ticket and related Pull-Requests
- https://ticketsystem.dbildungscloud.de/browse/BC-2364
- https://github.com/hpi-schul-cloud/nuxt-client/pull/2313

## Changes
- toast with a message about an already registered user
- translations for toast

## Screenshots of UI changes
Fixed

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by sszafGCA — DISMISSED** [review]

**Review by sszafGCA — APPROVED** [review]

File: src/pages/administration/StudentOverview.page.vue [hunk 1]
```diff
@@ -502,9 +503,15 @@ export default {
...
 				});
-				this.$toast.success(
-					this.$tc("pages.administration.sendMail.success", rowIds.length)
-				);
+				if (this.registrationLinks.totalMailsSend === rowIds.length) {
+					this.$toast.success(
+						this.$tc("pages.administration.sendMail.success", rowIds.length)
+					);
+				} else {
+					this.$toast.info(
```
Comment (Metauriel) [inline src/pages/administration/StudentOverview.page.vue:511]: do we know that this is the only issue that might arise? Do we have a way of interpreting the actual error from the backend, and do some handling based on that? (eg. use this message if user is already registered, and a general error message if something else went wrong)

**Review by Metauriel — CHANGES_REQUESTED** [review]

File: src/pages/administration/StudentOverview.page.vue (hunk 1 above)
Comment (MateuszSapalaPassCon) [reply src/pages/administration/StudentOverview.page.vue:511]: Example response from backend:
{
   "totalReceivedIds":[
      {
         "_id":"6370ed01b3a28164d4736396",
         "roles":[
            {
               "_id":"0000d186816abba584714c99",
               "name":"student",
               "updatedAt":"2022-05-20T13:33:18.077Z",
               "createdAt":"2017-01-01T00:06:37.148Z",
               "roles":[
                  "0000d186816abba584714c95"
               ],
               "permissions":[
                  "TASK_DASHBOARD_VIEW_V3",
                  "JOIN_MEETING",
                  "TEAM_CREATE",
                  "TEAM_EDIT"
               ],
               "__v":0,
               "displayName":"Schüler",
               "id":"0000d186816abba584714c99"
            }
         ],
         "firstName":"Mateusz",
         "lastName":"Sapala",
         "email":"example@gmail.com",
         "schoolId":"5f2987e020834114b8efd6f8",
         "preferences":{
            "registrationMailSend":true
         },
         "fullName":"Mateusz Sapala",
         "id":"6370ed01b3a28164d4736396"
      }
   ],
   "totalMailsSend":1,
   "alreadyRegisteredUsers":null
}

Backend code: https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/src/services/user/services/MailRegistrationLink.js#L85-L89
Don't know what author meant but it probably should look like:
return {
        totalReceivedIds: mailUsers.length,
	totalMailsSend,
	alreadyRegisteredUsers: mailUsers.length - totalMailsSend,
};

Than response for unregistered user would look like:
{
   "totalReceivedIds":1
   "totalMailsSend":1,
   "alreadyRegisteredUsers":0
}
And for registered user like :
{
   "totalReceivedIds":1
   "totalMailsSend":0,
   "alreadyRegisteredUsers":1
}

I can write fix for this backend and I can change error handling to base on "already Registered Users", but it'll be basically the same logic as it is. I don't see any further improvements as the backend doesn't return any other useful information. @Metauriel should I write fix for backend and if so should it be a separate ticket?

Comment (MateuszSapalaPassCon) [reply src/pages/administration/StudentOverview.page.vue:511]: Also if something else happened in the backend causing it to return a unsuccessfull return code, nuxt will show toast with a generic error message "Registration link(s) could not be sent" (https://github.com/hpi-schul-cloud/nuxt-client/blob/BC-2364-student- registration-already done-toast/src/pages/administration/StudentOverview.page.vue#L515-L520)

Comment (Metauriel) [reply src/pages/administration/StudentOverview.page.vue:511]: if other errors result in a different message, its fine for me

**Review by Metauriel — APPROVED** [review]

**Pull request merged**